### DISCUSSION TITLE
How fzf works with pipes?

### DISCUSSION THREAD

**Discussion by Kerruba** [junegunn/fzf#3023] [Q&A]
Hello,

I wanted to better understand how fzf is able to deal so well with piping. Many other projects (github CLI, bubbletea, etc.) seems to have some issues when is time to read stdin from pipe and pipe stdout to another command, while fzf instead works fine - and I think is one of is biggest powers.
I was trying to inspect the code to exactly understand how this behavior is implemented, but had some issues understading where to look. 
Could anyone give me some hints where to look or how this is done?

Thanks

**Comment by junegunn** [comment]
See https://github.com/junegunn/fzf/discussions/2686
